blockchain/blockdao/blockdao.go

```diff
@@ -48,6 +48,7 @@ type (
 		GetActionByActionHash(hash.Hash256, uint64) (action.SealedEnvelope, uint32, error)
 		GetReceiptByActionHash(hash.Hash256, uint64) (*action.Receipt, error)
 		DeleteBlockToTarget(uint64) error
+		StartingHeight() uint64
```

Comment:
`StartingHeight` feels more like a concept belong to the blockchain, so prefer to have this in `Blockchain` module, simply check `blockdao`'s height at the end of Start:
```
func (bc *blockchain) Start() {
// current code
bc.startHeight, err = bc.dao.Height()
}
```